src/stores/auth.js

```diff
@@ -117,10 +116,12 @@ export const useAuthStore = defineStore('auth', {
           const { gameId } = route.params;
           gameStore.requestSpectate(Number(gameId));
         }
-
         if (gameId && (isGame || isLobby)) {
           await this.requestReauthenticate({ username }).then(({ game }) => {
             gameStore.updateGame(game);
+            if (+router.currentRoute.value.params.gameId !== game.id) {
+              router.push(`${isGame ? '/game/' : '/lobby/'}${game.id}`);
```

Comment:
I'm wondering if this check + navigation is redundant with the socket-driven navigation fix. Just curious if the tests pass without it. That said it's a more general catch all for "server thinks you're in a different game than you are looking at; fix it", so I think it's a good idea

Reply:
The test do not pass without it, you enter the correct game, but the router does not see it.